### PR TITLE
Be/feature/ri 5105 wrapper for pub sub

### PR DESCRIPTION
**Pull request by irinaP-local**

### CODE REVIEW

File: redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts [hunk 1]
```diff
@@ -95,4 +95,45 @@ export class ClusterNodeRedisClient extends NodeRedisClient {
...
+  async unsubscribe(channel: string): Promise<void> {
+    await this.client.unsubscribe(channel);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async pUnsubscribe(channel: string): Promise<void> {
+    await this.client.pUnsubscribe(channel);
+  }
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts:138]: Can we put this logic to abstract ioredis.client class to not duplicate it for standalone, cluster and sentinel?

File: redisinsight/api/src/modules/redis/client/node-redis/standalone.node-redis.client.ts [hunk 2]
```diff
@@ -63,4 +63,45 @@ export class StandaloneNodeRedisClient extends NodeRedisClient {
...
+  async unsubscribe(channel: string): Promise<void> {
+    return this.client.unsubscribe(channel);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async pUnsubscribe(channel: string): Promise<void> {
+    return this.client.pUnsubscribe(channel);
+  }
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/redis/client/node-redis/standalone.node-redis.client.ts:106]: the same. let's put it to abstract class

**Review by arthosofteq — COMMENTED** [review]
lgtm but requires small changes

File: redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts [hunk 3]
```diff
@@ -49,10 +49,13 @@ export class ClusterNodeRedisClient extends NodeRedisClient {
...
    */
-  async sendPipeline(commands: RedisClientCommand[]): Promise<Array<[Error | null, RedisClientCommandReply]>> {
+  async sendPipeline(
+    commands: RedisClientCommand[],
+    options?: IRedisClientCommandOptions,
+  ): Promise<Array<[Error | null, RedisClientCommandReply]>> {
     return Promise.all(
       commands.map(
-        (cmd) => this.sendCommand(cmd)
+        (cmd) => this.sendCommand(cmd, options)
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts:58]: As I remember we must have some kind of `prepareCommandOptions` method to make our custom options match with native lib options. please use it

File: redisinsight/api/src/modules/redis/client/node-redis/standalone.node-redis.client.ts [hunk 4]
```diff
@@ -19,10 +19,13 @@ export class StandaloneNodeRedisClient extends NodeRedisClient {
...
    */
-  async sendPipeline(commands: RedisClientCommand[]): Promise<Array<[Error | null, RedisClientCommandReply]>> {
+  async sendPipeline(
+    commands: RedisClientCommand[],
+    options?: IRedisClientCommandOptions,
+  ): Promise<Array<[Error | null, RedisClientCommandReply]>> {
     return Promise.all(
       commands.map(
-        (cmd) => this.sendCommand(cmd)
+        (cmd) => this.sendCommand(cmd, options)
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/redis/client/node-redis/standalone.node-redis.client.ts:28]: the same

File: redisinsight/api/test/api/cli/POST-databases-id-cli-uuid-send_command.test.ts [hunk 5]
```diff
@@ -1008,8 +1008,12 @@ describe('POST /databases/:instanceId/cli/:uuid/send-command', () => {
...
         },
         checkFn: ({ body }) => {
-          expect(body.response).to.be.an('object');
-          expect(body.response).to.deep.eql({[constants.TEST_HASH_FIELD_1_NAME]: constants.TEST_HASH_FIELD_1_VALUE});
+          expect([
+            // result from ioredis
+            {[constants.TEST_HASH_FIELD_1_NAME]: constants.TEST_HASH_FIELD_1_VALUE},
+            // result from node-redis
+            [constants.TEST_HASH_FIELD_1_NAME, constants.TEST_HASH_FIELD_1_VALUE]
+          ]).to.deep.contain(body.response)
```
Comment (arthosofteq) [inline redisinsight/api/test/api/cli/POST-databases-id-cli-uuid-send_command.test.ts:1016]: let's add "todo" here. We should not forget to investigate this

File: redisinsight/api/test/api/cli/POST-databases-id-cli-uuid-send_command.test.ts [hunk 6]
```diff
@@ -1091,10 +1095,12 @@ describe('POST /databases/:instanceId/cli/:uuid/send-command (MULTI)', () => {
...
-          expect(body.response).to.deep.eq([
-            'OK',
-            'ReplyError: ERR value is not an integer or out of range',
-          ]);
+          expect([
+            // result from ioredis
+            ['OK', 'ReplyError: ERR value is not an integer or out of range'],
+            // result from node-redis
+            ['OK', 'Error: ERR value is not an integer or out of range'],
+          ]).to.deep.contain(body.response)
```
Comment (arthosofteq) [inline redisinsight/api/test/api/cli/POST-databases-id-cli-uuid-send_command.test.ts:1103]: the same

File: redisinsight/api/test/api/workbench/POST-databases-id-workbench-command_executions.test.ts [hunk 7]
```diff
@@ -381,14 +381,18 @@ describe('POST /databases/:instanceId/workbench/command-executions', () => {
...
+                response: {[constants.TEST_HASH_FIELD_1_NAME]: constants.TEST_HASH_FIELD_1_VALUE},
                 status: 'success',
-              },
-            ]);
+              }],
+              // result from node-redis
+              [{
+                response: [constants.TEST_HASH_FIELD_1_NAME, constants.TEST_HASH_FIELD_1_VALUE],
+                status: 'success',
+              }]
```
Comment (arthosofteq) [inline redisinsight/api/test/api/workbench/POST-databases-id-workbench-command_executions.test.ts:394]: the same

File: redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts (hunk 3 above)
Comment (irinaP-local) [reply redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts:58]: Pipeline calls a sendCommand that has preparation of options 
`return this.client.sendCommand(
      NodeRedisClient.prepareCommandArgs(command),
      NodeRedisClient.prepareCommandOptions(options),
    );`

Comment (arthosofteq) [reply redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts:58]: agree. forgot about it

**Review by arthosofteq — APPROVED** [review]

**Pull request merged**